### PR TITLE
Add a watch on CAPI cluster in NutanixMachineReconciler

### PR DESCRIPTION
**Pull request by thunderboltsid**
- Refactor SetupWithManager to call `Build` explicitly to add a watch to the controller after `Build` is called instead of calling `Complete`.
- Add a watch to the CAPI cluster with a `ClusterUnpausedAndInfrastructureReady` predicate
- Ensure `RateLimiter` gets correctly propagated to the underlying `controller.Options`. Earlier `controllerConfig.Ratelimiter` was not passed in the `controller.Options` struct passed to the controller builder

### CODE REVIEW

**Comment by thunderboltsid** [comment]
Change seems to be failing on lint issues that have been fixed in https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/437

**Review by adiantum — APPROVED** [review]
/lgtm
/approve

**Comment by deepakm-ntnx** [comment]
> * Ensure `RateLimiter` gets correctly propagated to the underlying `controller.Options`

Can we document how did we come to conclusion that this was not propagating correctly?

**Comment by thunderboltsid** [comment]
> > * Ensure `RateLimiter` gets correctly propagated to the underlying `controller.Options`
> 
> Can we document how did we come to conclusion that this was not propagating correctly?

By looking at the code and realizing that the controllerconfig.Ratelimiter was not passed in the controller.Options struct passed to the controller builder.

File: controllers/nutanixmachine_controller_test.go [hunk 1]
```diff
@@ -228,3 +236,133 @@ func TestNutanixMachineReconciler(t *testing.T) {
...
+		controllerConfig: &ControllerConfig{
+			MaxConcurrentReconciles: 1,
+		},
+	}
+
+	err = reconciler.SetupWithManager(ctx, mgr)
+	assert.Error(t, err)
+}
+
+func TestNutanixMachineReconciler_SetupWithManager_ClusterToTypedObjectsMapperError(t *testing.T) {
```
Comment (deepakm-ntnx) [inline controllers/nutanixmachine_controller_test.go:326]: comments on the purpose of the test would be appreciated.

**Review by deepakm-ntnx — APPROVED** [review]

File: controllers/nutanixmachine_controller_test.go (hunk 1 above)
Comment (thunderboltsid) [reply controllers/nutanixmachine_controller_test.go:326]: it's a test for the case when ClusterToTypedObjectsMapper call fails inside SetupWithManager. Largely added to appease the codecov check.

**Pull request merged**